src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -525,27 +524,31 @@ YlmSpherepack::first_and_second_derivative(
   return result;
 }
 
-YlmSpherepack::InterpolationInfoPerPoint::InterpolationInfoPerPoint(
+template <typename T>
+YlmSpherepack::InterpolationInfo<T>::InterpolationInfo(
     const size_t m_max, const std::vector<double>& pmm,
-    const std::array<double, 2>& target)
+    const std::array<T, 2>& target)
     : cos_theta(cos(target[0])),
       cos_m_phi(m_max + 1),
       sin_m_phi(m_max + 1),
-      pbar_factor(m_max + 1) {
-  const double sin_theta = sin(target[0]);
-  const double phi = target[1];
+      pbar_factor(m_max + 1),
+      num_points(get_size(target[0])) {
+  // do not use auto here, the blaze expression templates will create a dangling
+  // reference that is not caught
+  const T sin_theta = sin(target[0]);
```

Comment:
This code just got _really_ expensive. You now have `4 + m_max-1`  additional memory allocations, that's pretty brutal. You should try to get that down a lot... For example, the memory allocations on lines 550 and 551 should be outside the for loop. It would be even better if you could use a `Variables` in the case you have a `DataVector` so you only do one allocation. Having said that, with some clever trickery I think you can eliminate most memory allocations. For example, you could reuse the allocations in the for loop that you need for `cos_m_phi[m]` and `sin_m_phi[m]`, I think.